applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -214,40 +176,103 @@ void FindNeighbourElementsOfConditionsProcess::Execute()
                     if (itFace != FacesMap.end()) {
                         // condition is found!
                         // but check if there are more than one condition on the element
-                        std::pair <hashmap::iterator, hashmap::iterator> ret;
-                        ret = FacesMap.equal_range(itFace->first);
-                        for (hashmap::iterator it=ret.first; it!=ret.second; ++it) {
-                            std::vector<Condition::Pointer>& ListConditions = it->second;
-
-                            GlobalPointersVector< Element > VectorOfNeighbours;
-                            VectorOfNeighbours.resize(1);
-                            VectorOfNeighbours(0) = Element::WeakPointer( *itElem.base() );
-
-                            for (Condition::Pointer pCondition : ListConditions) {
-                                pCondition->Set(VISITED,true);
-                                pCondition->SetValue(NEIGHBOUR_ELEMENTS, VectorOfNeighbours);
-                            }
-                        }
+                        CheckForMultipleConditionsOnElement(FacesMap, itFace, itElem);
                     }
                 }
             }
         }
     }
 
+
+    //check that all of the conditions belong to at least an element.
+    all_conditions_visited = CheckIfAllConditionsAreVisited();
+
+    if (all_conditions_visited) {
+        // if all conditions are found, no need for further checks:
+        return;
+    }
+
+    // check 1D elements, note that this has to happen after procedures to find 2 and 3d neighbours are alredy performed, such that 1D elements are only added
+    // as neighbours when the condition is not neighbouring 2D or 3D elements
+    this->CheckIf1DElementIsNeighbour(FacesMap);
+    
+    
     //check that all of the conditions belong to at least an element. Throw an error otherwise (this is particularly useful in mpi)
-    AllVisited = true;
+    all_conditions_visited = true;
     for (auto& rCond : mrModelPart.Conditions()) {
         if (rCond.IsNot(VISITED)) {
-            AllVisited = false;
+            all_conditions_visited = false;
             KRATOS_INFO("Condition without any corresponding element, ID ") << rCond.Id() << std::endl;
         }
     }
 
-    KRATOS_ERROR_IF_NOT(AllVisited) << "Some conditions found without any corresponding element" << std::endl;
+    KRATOS_ERROR_IF_NOT(all_conditions_visited) << "Some conditions found without any corresponding element" << std::endl;
 
     KRATOS_CATCH("")
 }
 
+bool FindNeighbourElementsOfConditionsProcess::CheckIfAllConditionsAreVisited() const
+{
+    for (auto& r_cond : mrModelPart.Conditions()) {
+        if (r_cond.IsNot(VISITED)) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
I believe this is an algorithm from the STL:
```suggestion
    const auto& rConditions = mrModelPart.Conditions();
    return std::all_of(rConditions.begin(), rConditions.end(),
                       [](const auto& r_cond){return r_cond.Is(VISITED);});
```
Using such an algorithm makes your intent very clear and easy to read. But please check (and test) this suggested change carefully before incorporating it in your work.